File: src/main/java/phoenix/client/gui/diaryPages/elements/ADiaryElement.java
package phoenix.client.gui.diaryPages.elements;

import com.mojang.blaze3d.matrix.MatrixStack;
import net.minecraft.client.gui.FontRenderer;
import net.minecraft.client.gui.screen.inventory.ContainerScreen;
import net.minecraft.nbt.CompoundNBT;
import phoenix.containers.DiaryContainer;

public abstract class ADiaryElement
{
    abstract public int getHeight(int maxSizeXIn, int maxSizeYIn);
    abstract public void render(MatrixStack stack, ContainerScreen<DiaryContainer> gui, FontRenderer renderer, int xSize, int ySize, int x, int y, int depth);
    abstract public CompoundNBT serialize();

    @Override
    abstract public String toString();
}

File: src/main/kotlin/phoenix/client/gui/diaryPages/elements/ADiaryElement.kt
package phoenix.client.gui.diaryPages.elements

import com.mojang.blaze3d.matrix.MatrixStack
import net.minecraft.client.gui.FontRenderer
import net.minecraft.client.gui.screen.inventory.ContainerScreen
import net.minecraft.nbt.CompoundNBT
import phoenix.containers.DiaryContainer


abstract class ADiaryElement
{
    abstract fun getHeight(maxSizeXIn: Int, maxSizeYIn: Int): Int
    abstract fun render(
        stack: MatrixStack,
        gui: ContainerScreen<DiaryContainer>,
        renderer: FontRenderer,
        xSize: Int,
        ySize: Int,
        x: Int,
        y: Int,
        depth: Int
    )

    abstract fun serialize(): CompoundNBT
    abstract override fun toString(): String
}
